### PR TITLE
Kotlintest matchers

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Use some kotlintest matchers instead of manually checking the content of collections and maps.

### CODE REVIEW

File: scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt [hunk 1]
```diff
@@ -139,8 +141,7 @@ abstract class AbstractStorageTest : StringSpec() {
 
             result shouldBe ADD_RESULT_SUCCESS
             storedResults.id shouldBe id
-            storedResults.results.size shouldBe 1
-            storedResults.results[0] shouldBe scanResult
+            storedResults.results should containExactly(scanResult)
```
Comment (fviernau) [inline scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt]: I think in other places we use already the infix fun `shouldContainExactly` instead. How about using it here as well?

File: model/src/test/kotlin/ProjectAnalyzerResultTest.kt [hunk 2]
```diff
@@ -83,11 +85,13 @@ class ProjectAnalyzerResultTest : StringSpec({
             issues = listOf(issue7, issue8)
         )
 
-        val issues = result.collectIssues()
-        issues.size shouldBe 4
-        issues[Identifier("type:namespace:name:version")] shouldBe listOf(issue7, issue8)
-        issues[Identifier("type1:namespace1:name1:version1")] shouldBe listOf(issue3, issue4)
-        issues[Identifier("type2:namespace2:name2:version2")] shouldBe listOf(issue1, issue2)
-        issues[Identifier("type3:namespace3:name3:version3")] shouldBe listOf(issue5, issue6)
+        result.collectIssues() should containExactly(
```
Comment (fviernau) [inline model/src/test/kotlin/ProjectAnalyzerResultTest.kt]: While at it we could remove the `over assertion` about the ordering.

File: scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt (hunk 1 above)
Comment (fviernau) [reply scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt]: Hm or maybe the other way round, I like this one more. How about you?

File: model/src/test/kotlin/ProjectAnalyzerResultTest.kt (hunk 2 above)
Comment (mnonnenmacher) [reply model/src/test/kotlin/ProjectAnalyzerResultTest.kt]: I think the map matcher does not assume any order, at least there is no `containExactlyInAnyOrder` variant.

File: scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt (hunk 1 above)
Comment (mnonnenmacher) [reply scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt]: I think `should containExactly` read a bit better. Let's talk on Monday on which of both we want to align.

File: model/src/test/kotlin/ProjectAnalyzerResultTest.kt (hunk 2 above)
Comment (fviernau) [reply model/src/test/kotlin/ProjectAnalyzerResultTest.kt]: It's for maps, my bad! So no objections on this PR on my side.

File: scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt (hunk 1 above)
Comment (fviernau) [reply scanner/src/funTest/kotlin/storages/AbstractStorageTest.kt]: Agree. I think you can merge this.

**Pull request merged**